packages/vscode-extension/src/debugging/CDPSession.ts

```diff
@@ -286,6 +291,26 @@ export class CDPSession {
   }
 
   private async handleDebuggerPaused(message: any) {
+    if (!this.debugSessionReady) {
+      await new Promise((resolve, reject) => {
```

Comment:
I prefer `Promise.withResolvers()` approach as it is easier to follow and we are using it more and more often thorough the codebase.

Reply:
I felt like since we `await` it immediately, the constructor form made more sense, but maybe you're right.

Reply:
Extracted the "resolve a promise on event or reject after a timeout" part to a helper function, I think it reads better that way.